src/solvers/strings/string_refinement.cpp

```diff
@@ -1003,7 +1003,7 @@ static optionalt<exprt> get_array(
            << messaget::eom;
     stream << "(sr::get_array) this can also happen on invalid object access"
            << messaget::eom;
-    return nil_exprt();
+    return {};
```

Comment:
This can potentially change the behaviour in the test `jbmc/regression/jbmc-strings/long_string/test_abort.desc` because here MAX_CONCRETE_STRING_SIZE was exceeded (on purpose) and the value of the array replaced by nil in the trace. Can you check what happens now in the trace?

Reply:
Indeed that broke that test. I reverted that change and consequently changed the refactoring a bit. I moved the max length check back into `get_array`.